database/src/types.rs

```diff
@@ -49,96 +45,319 @@ use bytemuck::{AnyBitPattern, NoUninit, Pod, Zeroable};
+/// let c2: Cow<'_, BlockInfoV2> = Storable::from_bytes_unaligned(b);
+/// assert_eq!(&a, c);
+/// assert_eq!(c, c2.as_ref());
+/// ```
+///
+/// # Size & Alignment
+/// ```rust
+/// # use cuprate_database::types::*;
+/// # use std::mem::*;
+/// assert_eq!(size_of::<BlockInfoV2>(), 72);
+/// assert_eq!(align_of::<BlockInfoV2>(), 8);
+/// ```
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash, Pod, Zeroable)]
+#[repr(C)]
+pub struct BlockInfoV2 {
+    /// TODO
+    pub timestamp: u64,
+    /// TODO
+    pub total_generated_coins: u64,
+    /// TODO
+    pub weight: u64,
+    /// TODO
+    pub block_hash: [u8; 32],
+    /// TODO
+    pub cumulative_difficulty: u64,
+    /// TODO
     ///
-    /// TODO: this is basically leeway to
-    /// add more things to our structs too,
-    /// because otherwise this space is wasted.
-    pub _pad: [u8; 7],
+    /// TODO: note that this is originally u32,
+    /// but is u64 here for padding reasons.
+    pub cumulative_rct_outs: u64,
 }
 
-//---------------------------------------------------------------------------------------------------- TestType2
-/// TEST2
+//---------------------------------------------------------------------------------------------------- BlockInfoV3
+/// TODO
 ///
 /// ```rust
+/// # use std::borrow::*;
 /// # use cuprate_database::{*, types::*};
-/// // Assert bytemuck is correct.
-/// let a = TestType2 { u: 1, b: [1; 32] }; // original struct
-/// let b = bytemuck::must_cast::<TestType2, [u8; 40]>(a); // cast into bytes
-/// let c = bytemuck::must_cast::<[u8; 40], TestType2>(b); // cast back into struct
-/// assert_eq!(a, c);
-/// assert_eq!(c.u, 1);
-/// assert_eq!(c.b, [1; 32]);
+/// // Assert Storable is correct.
+/// let a = BlockInfoV3 {
+///     timestamp: 1,
+///     total_generated_coins: 123,
+///     weight: 321,
+///     cumulative_difficulty_low: 111,
+///     cumulative_difficulty_high: 112,
+///     block_hash: [54; 32],
+///     cumulative_rct_outs: 2389,
+///     long_term_weight: 2389,
+/// };
+/// let b = Storable::as_bytes(&a);
+/// let c: &BlockInfoV3 = Storable::from_bytes(b);
+/// let c2: Cow<'_, BlockInfoV3> = Storable::from_bytes_unaligned(b);
+/// assert_eq!(&a, c);
+/// assert_eq!(c, c2.as_ref());
+/// ```
 ///
+/// # Size & Alignment
+/// ```rust
+/// # use cuprate_database::types::*;
+/// # use std::mem::*;
+/// assert_eq!(size_of::<BlockInfoV3>(), 88);
+/// assert_eq!(align_of::<BlockInfoV3>(), 8);
+/// ```
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Copy, Clone, Debug, PartialEq, PartialOrd, Eq, Ord, Hash, Pod, Zeroable)]
+#[repr(C)]
+pub struct BlockInfoV3 {
+    /// TODO
+    pub timestamp: u64,
+    /// TODO
+    pub total_generated_coins: u64,
+    /// TODO
+    pub weight: u64,
+    // Maintain 8 byte alignment.
+    /// TODO
+    pub cumulative_difficulty_low: u64,
+    /// TODO
+    pub cumulative_difficulty_high: u64,
```

Comment:
For the next PR where a maximum alignment isn't needed this can be changed back into a u128